components/events/Event.vue

```diff
@@ -0,0 +1,123 @@
+<script setup lang="ts">
+import IconDelete from "./IconDelete.vue"
+import IconEdit from "./IconEdit.vue"
+import IconLocation from "./IconLocation.vue"
+import IconSpeaker from "./IconSpeaker.vue"
+import IconTime from "./IconTime.vue"
+import { getEvent, deletePost, Event } from "./utils"
+
+
+const p = defineProps(["id"])
+
+const thisEvent: Ref<Event> = ref({
```

Comment:
no need to type the var manually, just do `ref<Event>(.....)`

Reply:
also, i would set it to null in the beginning and only display everything with a `v-if="thisEvent"`